cpp/RowGroupMetaData.cpp

```diff
@@ -32,4 +31,58 @@ extern "C"
 	{
 		TRYCATCH(*total_byte_size = row_group_meta_data->total_byte_size();)
 	}
+
+	PARQUETSHARP_EXPORT ExceptionInfo* RowGroupMetaData_Sorting_Columns(const RowGroupMetaData* row_group_meta_data, int32_t** column_indices, bool** descending, bool** nulls_first, int* num_columns)
+	{
+		try
+		{
+			auto sorting_columns = row_group_meta_data->sorting_columns();
+			
+			*num_columns = static_cast<int>(sorting_columns.size());
+			
+			if (*num_columns > 0)
+			{
+				*column_indices = new int32_t[*num_columns];
+				*descending = new bool[*num_columns];
+				*nulls_first = new bool[*num_columns];
+				
+				for (int i = 0; i < *num_columns; ++i)
+				{
+					(*column_indices)[i] = sorting_columns[i].column_idx;
+					(*descending)[i] = sorting_columns[i].descending;
+					(*nulls_first)[i] = sorting_columns[i].nulls_first;
+				}
+			}
+			else
+			{
+				*column_indices = nullptr;
+				*descending = nullptr;
+				*nulls_first = nullptr;
+			}
+			
+			return nullptr;
+		}
+		catch (const std::exception& e)
+		{
+			return new ExceptionInfo(typeid(e).name(), e.what());
+		}
+	}
+
+	PARQUETSHARP_EXPORT void RowGroupMetaData_Sorting_Columns_Free(int32_t* column_indices, bool* descending, bool* nulls_first)
+	{
+		if (column_indices != nullptr)
```

Comment:
I think that the null checks are not needed, as deleting a null pointer has no effect.